### ISSUE TITLE
SyntaxError excpetion in response of requestSync is not rejected by promise

### ISSUE THREAD

**Issue by AiDirex** [actions-on-google/actions-on-google-nodejs#363]
If response of the call to requestSync method is not in proper JSON then the response handler throws SyntaxError exception at [JSON.parse](https://github.com/actions-on-google/actions-on-google-nodejs/blob/1c9c56b1e837a7cc54bd1c8058d3ddbe2747863b/src/service/smarthome/smarthome.ts#L301), without rejecting the promise causing the whole node application to crash.
it should be wrapped around a try-catch block and rejected properly.

**Comment by Fleker** [comment]
So you're calling requestSync with some data, but for some reason it fails and it sends you a non-JSON body?

**Comment by AiDirex** [comment]
@Fleker yes, I saw it in the logs a few times. I saw 403 unauthorized errors.
Anyway, there certainly is a chance of it failling thus it should be wrapped around try-catch. for example google service not responding properly, being down or if you proxied the request and the proxy being down, .... 
this should not take my whole application down.
